### PR TITLE
[kubernetes] Enable three new fuzz targets (Duration, MicroTime, Time).

### PR DESCRIPTION
**Pull request by Dor1s**
This depends on https://github.com/kubernetes/kubernetes/pull/84168

//cc @BrendanSChang @mikedanese please ping this issue when the upstream PR is merged

### CODE REVIEW

File: projects/kubernetes/build.sh [hunk 1]
```diff
@@ -33,7 +33,10 @@ function compile_fuzzer {
   $CXX $CXXFLAGS $LIB_FUZZING_ENGINE "${fuzzer}.a" -lpthread -o "${OUT}/${fuzzer}"
 }
 
+compile_fuzzer "yaml" "FuzzDuration"
+compile_fuzzer "yaml" "FuzzMicroTime"
```
Comment (BrendanSChang) [inline projects/kubernetes/build.sh]: Can you change these to FuzzDurationStrict and FuzzMicroTimeStrict?

File: projects/kubernetes/build.sh [hunk 2]
```diff
@@ -33,7 +33,10 @@ function compile_fuzzer {
...
 compile_fuzzer "yaml" "FuzzSigYaml"
+compile_fuzzer "yaml" "FuzzTime"
```
Comment (BrendanSChang) [inline projects/kubernetes/build.sh]: And this to FuzzTimeStrict?

**Comment by BrendanSChang** [comment]
> Thanks for adding these! [kubernetes/kubernetes#84168](https://github.com/kubernetes/kubernetes/pull/84168) has been merged but I actually had a small typo that's pending in [kubernetes/kubernetes#84409](https://github.com/kubernetes/kubernetes/pull/84409). I'll update this once that PR has been merged.

The dependencies have been merged in, so this is good to go once the comments above are resolved. Thanks!

File: projects/kubernetes/build.sh (hunk 1 above)
Comment (Dor1s) [reply projects/kubernetes/build.sh]: Done

File: projects/kubernetes/build.sh (hunk 2 above)
Comment (Dor1s) [reply projects/kubernetes/build.sh]: Done

**Comment by Dor1s** [comment]
Hm, looks like all these three targets have a startup crash:

```
Broken fuzz targets (3):
yaml_FuzzDurationStrict:
BAD BUILD: /out/yaml_FuzzDurationStrict seems to have either startup crash or exit:
INFO: Seed: 1337
INFO: 65536 Extra Counters
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
panic: result != input
goroutine 17 [running, locked to thread]:
k8s.io/kubernetes/test/fuzz/yaml.FuzzDurationStrict(0x602000001090, 0x0, 0x0, 0x7ffd2620c478)
	/gopath/src/k8s.io/kubernetes/test/fuzz/yaml/yaml.go:45 +0x2d9
main.LLVMFuzzerTestOneInput(0x602000001090, 0x0, 0xfdf960)
	k8s.io/kubernetes/test/fuzz/yaml/go.fuzz.main/main.go:35 +0x66
main._cgoexpwrap_cf0b73e90b38_LLVMFuzzerTestOneInput(0x602000001090, 0x0, 0x13b9200)
	_cgo_gotypes.go:64 +0x37
AddressSanitizer:DEADLYSIGNAL
=================================================================
==120==ERROR: AddressSanitizer: ABRT on unknown address 0x000000000078 (pc 0x0000005b0611 bp 0x10c0000d7c58 sp 0x10c0000d7c40 T0)
SCARINESS: 10 (signal)
    #0 0x5b0610 in runtime.raise runtime/sys_linux_amd64.s:149
AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: ABRT runtime/sys_linux_amd64.s:149 in runtime.raise
==120==ABORTING
MS: 0 ; base unit: 0000000000000000000000000000000000000000
artifact_prefix='./'; Test unit written to ./crash-da39a3ee5e6b4b0d3255bfef95601890afd80709
Base64: 
yaml_FuzzMicroTimeStrict:
BAD BUILD: /out/yaml_FuzzMicroTimeStrict seems to have either startup crash or exit:
INFO: Seed: 1337
INFO: 65536 Extra Counters
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
panic: result != input
goroutine 17 [running, locked to thread]:
k8s.io/kubernetes/test/fuzz/yaml.FuzzMicroTimeStrict(0x602000001090, 0x0, 0x0, 0x7ffc38323a58)
	/gopath/src/k8s.io/kubernetes/test/fuzz/yaml/yaml.go:65 +0x2d9
main.LLVMFuzzerTestOneInput(0x602000001090, 0x0, 0xfdf940)
	k8s.io/kubernetes/test/fuzz/yaml/go.fuzz.main/main.go:35 +0x66
main._cgoexpwrap_63d8d824bc24_LLVMFuzzerTestOneInput(0x602000001090, 0x0, 0x13b9200)
	_cgo_gotypes.go:64 +0x37
AddressSanitizer:DEADLYSIGNAL
=================================================================
==242==ERROR: AddressSanitizer: ABRT on unknown address 0x0000000000f2 (pc 0x0000005b0611 bp 0x10c0000c1c58 sp 0x10c0000c1c40 T0)
SCARINESS: 10 (signal)
    #0 0x5b0610 in runtime.raise runtime/sys_linux_amd64.s:149
AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: ABRT runtime/sys_linux_amd64.s:149 in runtime.raise
==242==ABORTING
MS: 0 ; base unit: 0000000000000000000000000000000000000000
artifact_prefix='./'; Test unit written to ./crash-da39a3ee5e6b4b0d3255bfef95601890afd80709
Base64: 
yaml_FuzzTimeStrict:
BAD BUILD: /out/yaml_FuzzTimeStrict seems to have either startup crash or exit:
INFO: Seed: 1337
INFO: 65536 Extra Counters
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
panic: result != input
goroutine 17 [running, locked to thread]:
k8s.io/kubernetes/test/fuzz/yaml.FuzzTimeStrict(0x602000001090, 0x0, 0x0, 0x7ffe43933a28)
	/gopath/src/k8s.io/kubernetes/test/fuzz/yaml/yaml.go:99 +0x2d3
main.LLVMFuzzerTestOneInput(0x602000001090, 0x0, 0xfdf940)
	k8s.io/kubernetes/test/fuzz/yaml/go.fuzz.main/main.go:35 +0x66
main._cgoexpwrap_a32834f4ff63_LLVMFuzzerTestOneInput(0x602000001090, 0x0, 0x13b9200)
	_cgo_gotypes.go:64 +0x37
AddressSanitizer:DEADLYSIGNAL
=================================================================
==363==ERROR: AddressSanitizer: ABRT on unknown address 0x00000000016b (pc 0x0000005b0611 bp 0x10c0000c1c58 sp 0x10c0000c1c40 T0)
SCARINESS: 10 (signal)
    #0 0x5b0610 in runtime.raise runtime/sys_linux_amd64.s:149
AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: ABRT runtime/sys_linux_amd64.s:149 in runtime.raise
==363==ABORTING
MS: 0 ; base unit: 0000000000000000000000000000000000000000
artifact_prefix='./'; Test unit written to ./crash-da39a3ee5e6b4b0d3255bfef95601890afd80709
Base64: 
ERROR: 42% of fuzz targets seem to be broken. See the list above for a detailed information.
Check build failed.
```

You can reproduce it locally by following https://google.github.io/oss-fuzz/advanced-topics/reproducing/#reproducing-build-failures

**Comment by BrendanSChang** [comment]
Ah, sorry...I fixed the validation in kubernetes/kubernetes#84480. I'll let you know when it's merged in. I tested it locally so it should work:
```
$ python infra/helper.py check_build --sanitizer address --engine libfuzzer kubernetes             
Running: docker run --rm -i --privileged -e FUZZING_ENGINE=libfuzzer -e SANITIZER=address -e ARCHITECTURE=x86_64 -v /usr/local/google/home/brendanschang/oss-fuzz/build/out/kubernetes:/out -t gcr.io/oss-fuzz-base/base-runner test_all                                               
INFO: performing bad build checks for /out/yaml_FuzzMicroTimeStrict.
INFO: performing bad build checks for /out/yaml_FuzzTimeStrict.
INFO: performing bad build checks for /out/json_FuzzStrictDecode.
INFO: performing bad build checks for /out/yaml_FuzzYamlV2.
INFO: performing bad build checks for /out/json_FuzzNonStrictDecode.
INFO: performing bad build checks for /out/yaml_FuzzDurationStrict.
INFO: performing bad build checks for /out/yaml_FuzzSigYaml.
7 fuzzers total, 0 seem to be broken (0%).
Check build passed.
```

**Comment by Dor1s** [comment]
Re-running Travis as the upstream CL is merged.

**Pull request merged**